src/components/ExperienceTimeline.tsx

```diff
@@ -1,23 +1,77 @@
 import { Box, Text } from "@chakra-ui/react";
-import { ExperienceDetail } from "./Experience";
-import styles from "./ExperienceTimeline.module.css";
+import { Experience } from "../consts";
 
 interface ExperienceTimelineProps {
-  experienceDetails: ExperienceDetail[];
-  updateExperienceDataIndex: (index: number) => void;
+  experienceDetails: Experience[];
+  setExperience: (experience: Experience) => void;
 }
 
+const getPseudoElementStyles = (isHovered: boolean) => {
+  let width = "0.313rem";
+  let height = "0.313rem";
+  let top = "0.625rem";
+  let left = "-0.05rem";
+  let boxShadow =
+    "0 0 0 4px grey, inset 0 2px 0 rgb(0 0 0 / 8%), 0 3px 0 4px rgb(0 0 0 / 5%)";
+  let background = "black";
+
+  if (isHovered) {
+    width = "0.5rem";
+    height = "0.5rem";
+    top = "0.75rem";
+    left = "-0.15rem";
+    boxShadow =
+      "0 0 0 4px black, inset 0 2px 0 rgb(0 0 0 / 8%), 0 3px 0 4px rgb(0 0 0 / 5%)";
+    background = "white";
+  }
+
+  return {
+    content: `""`,
+    position: "absolute",
+    width,
+    height,
+    top,
+    left,
+    borderRadius: "50%",
+    zIndex: "1",
+    boxShadow,
+    background,
+    transition: "all 0.3s ease-in-out",
+    cursor: "pointer",
+  };
+};
+
 const ExperienceTimeline = ({
   experienceDetails,
-  updateExperienceDataIndex,
+  setExperience,
 }: ExperienceTimelineProps) => {
   return (
-    <Box className={styles.timeline}>
-      {experienceDetails.map((experience, index) => (
+    <Box
+      position="relative"
```

Comment:
Please check if we can separate these Box components with separate custom components with these styles embedded to avoid cluttering of component logic with component style. If not then, maybe we can define separate objects for these and then spread those here in the Box component